### PR TITLE
[UITests] CollectionView

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Add set of cross platform UITests tests for CollectionView galleries 

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ItemsSourceGenerator.cs [hunk 1]
```diff
@@ -23,7 +23,8 @@ public CollectionViewGalleryTestItem(DateTime date, string caption, string image
 
 		public override string ToString()
 		{
-			return $"{Date:D}";
+			return $"Item: {Index}";
+			//return $"{Date:D}";
```
Comment (hartez) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ItemsSourceGenerator.cs]: Drop the comments if we don't need this.

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselCodeGallery.cs [hunk 2]
```diff
@@ -27,6 +27,7 @@ public CarouselCodeGallery(ItemsLayoutOrientation orientation)
 
 			var carouselView = new CarouselView
 			{
+				AutomationId = "collectionview",
```
Comment (hartez) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselCodeGallery.cs]: Should this be "carouselview"?

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CollectionViewGallery.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@ public CollectionViewGallery()
 			{
 				Children =
 				{
+					new Button { Text ="Enable CollectionView", AutomationId = "EnableCollectionView", Command = new Command(()=> Device.SetFlags(new[] { "CollectionView_Experimental" })) },
```
Comment (hartez) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CollectionViewGallery.cs]: Won't this blow away any flags which are already set? Will that be a problem if we want to test CV and Fast Renderers at the same time?

File: Xamarin.Forms.Core.UITests.Shared/Utilities/AppExtensions.cs [hunk 4]
```diff
@@ -28,8 +28,12 @@ public static void NavigateToGallery (this IApp app, string page)
 			var text = Regex.Match (page, "'(?<text>[^']*)'").Groups["text"].Value;
 
 			app.WaitForElement("SearchBar");
-			app.EnterText (q => q.Raw ("* marked:'SearchBar'"), text);
+			app.EnterText(q => q.Raw("* marked:'SearchBar'"), text);
 
+			//var element = app.Query("SearchBar").First();
```
Comment (hartez) [inline Xamarin.Forms.Core.UITests.Shared/Utilities/AppExtensions.cs]: Get rid of the commented code if we don't need it.

File: Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj [hunk 5]
```diff
@@ -50,7 +50,7 @@
     <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
     <PackageReference Include="NUnit" Version="2.6.4" />
     <PackageReference Include="Xam.Plugin.DeviceInfo" Version="3.0.2" />
-    <PackageReference Include="Xamarin.UITest" Version="2.2.7" />
+    <PackageReference Include="Xamarin.UITest" Version="2.2.7.2001-dev" />
```
Comment (hartez) [inline Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj]: Was this update intentional?

File: Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj [hunk 6]
```diff
@@ -91,7 +91,7 @@
   </Target>
   <Target Name="_CopyXamarinUITestFiles" AfterTargets="Build">
     <ItemGroup>
-      <_XamarinUITestFiles Include="$(NuGetPackageRoot)Xamarin.UITest\%(Version)\**" Condition="@(PackageReference -> '%(Identity)') == 'Xamarin.UITest'" InProject="False" />
+      <_XamarinUITestFiles Include="$(NuGetPackageRoot)Xamarin.UITest\%(Version)\**" Condition="@(PackageReference -&gt; '%(Identity)') == 'Xamarin.UITest'" InProject="False" />
```
Comment (hartez) [inline Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj]: Was this change intentional?

Comment (rmarinho) [reply Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj]: nop

File: Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj (hunk 5 above)
Comment (rmarinho) [reply Xamarin.Forms.Core.iOS.UITests/Xamarin.Forms.Core.iOS.UITests.csproj]: nop

File: Xamarin.Forms.Core.UITests.Shared/Utilities/AppExtensions.cs (hunk 4 above)
Comment (rmarinho) [reply Xamarin.Forms.Core.UITests.Shared/Utilities/AppExtensions.cs]: shouldn't had commit this was testing stuff on iOS

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CollectionViewGallery.cs (hunk 3 above)
Comment (rmarinho) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CollectionViewGallery.cs]: That's a good question , need to try it. Do you had other idea to set this ? I also restart the gallery after the test finishes so any flags would be set again for next tests.

Comment (paymicro) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CollectionViewGallery.cs]: maybe implement something like `Device.AddFlagIfNotExist` ? also remove...

File: Xamarin.Forms.Controls/GalleryPages/GalleryBuilder.cs [hunk 7]
```diff
@@ -6,7 +6,8 @@ public static class GalleryBuilder
 	{
 		public static Button NavButton(string galleryName, Func<ContentPage> gallery, INavigation nav)
 		{
-			var button = new Button { Text = $"{galleryName}" };
+			var automationId = galleryName.Replace(" ", "").Replace("(", "").Replace(")", "");
```
Comment (paymicro) [inline Xamarin.Forms.Controls/GalleryPages/GalleryBuilder.cs]: Maybe it's better to use a regular expression to remove all unnecessary characters at once
```suggestion
			var automationId = System.Text.RegularExpressions.Regex.Replace(galleryName, " |\\(|\\)", string.Empty);
```

**Comment by rmarinho** [comment]
iOS tests failing https://github.com/xamarin/Xamarin.Forms/issues/5019

**Comment by rmarinho** [comment]
can you review this again @hartez  thanks

**Comment by hartez** [comment]
> can you review this again @hartez thanks

@rmarinho I think this might still be wrong: https://github.com/xamarin/Xamarin.Forms/pull/4697#discussion-diff-240429915R30
